src/components/MaterialUI/DataDisplay/Divider.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import DividerMui from "@material-ui/core/Divider";
+import { makeStyles } from "@material-ui/core/styles";
+import DividerProps from "./dividerProps";
+import classNames from "classnames";
+
+const useStyles = makeStyles(theme => ({
+	dividerRoot: {
+		backgroundColor: theme.palette.grey.dark,
+	},
+	dividerLight: {
+		backgroundColor: theme.palette.grey.borders,
+	},
+	dividerVertical: {
+		height: theme.spacing(3),
+		margin: `0 ${theme.spacing(1)}`,
+	},
+}));
+
+const Divider = ({ dividerProps }) => {
+	const classes = useStyles();
+
+	if (dividerProps != null && dividerProps instanceof DividerProps === false) {
+		throw new TypeError("dividerProps property is not of type DividerProps");
+	}
+
+	const orientation = dividerProps?.get(DividerProps.propNames.orientation);
+	const light = dividerProps?.get(DividerProps.propNames.light);
+	const variant = dividerProps?.get(DividerProps.propNames.variant);
+
+	var customStyles = dividerProps?.get(DividerProps.propNames.classes);
```

Comment:
It's not necessary to have classes in your props. The idea is that instead of accessing classes props you can access styles directly (using getStyle). And to set style use setStyle.

Also replace var with const. So, do smth like

const rootStyle = divierProps?.getStyle(DividerProps.ruleNames.root.

Same for others (like lightStyle, verticalStyle, etc.)